ts/src/header-validator/source.test.ts

```diff
@@ -1927,6 +1930,223 @@ const testCases: TestCase[] = [
       },
     ],
   },
+
+  // Attribution Scope
+  {
+    name: 'attribution-scope-limit-negative',
+    json: `{
+      "destination": "https://a.test",
+      "trigger_data": [],
+      "attribution_scope_limit": -1
+    }`,
+    expectedErrors: [
+      {
+        path: ['attribution_scope_limit'],
+        msg: 'must fit in an unsigned 32-bit integer'
+      },
+    ],
+  },
+  {
+    name: 'attribution-scope-limit-zero',
```

Comment:
Please also add test cases for non-integer values and for exceeding `MAX_UINT32`.

Reply:
Done.